docs/content/guides/dagster-pipes/kubernetes.mdx

```diff
@@ -0,0 +1,248 @@
+
+In this step, you’ll create a Kubernetes container that runs some code that uses `dagster-pipes`.
+
+### Step 1.1: Write a Python script
+
+First, you'll write a Python script that uses `dagster-pipes` and is executed in a container via Kubernetes:
+
+```python
+# my_python_script.py
+
+from dagster_pipes import open_dagster_pipes
+
+with open_dagster_pipes() as pipes:
+    # Stream log message back to Dagster
+    pipes.log.info(f"Using some_parameter value: {some_parameter_value}")
+
+    # ... your code that computes and persists the asset
+
+    # Stream asset materialization metadata and data version back to Dagster.
+    # This should be called after you've computed and stored the asset value. We
+    # omit the asset key here because there is only one asset in scope, but for
+    # multi-assets you can pass an `asset_key` parameter.
+    pipes.report_asset_materialization(
+        metadata={
+            "some_metric": {"raw_value": some_parameter_value + 1, "type": "int"}
+        },
+        data_version="alpha",
+    )
+```
+
+Let's review what this code does:
+
+- Imports <PyObject object="open_dagster_pipes" module="dagster_pipes" /> from `dagster_pipes`
+
+- **Initializes the Dagster Pipes context (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), which yields an instance of <PyObject object="PipesContext" module="dagster_pipes" /> called `pipes`.**
+
+  We're using the default context loader (<PyObject object="PipesDefaultContextLoader" module="dagster_pipes" />) and message writer (<PyObject object="PipesDefaultMessageWriter" module="dagster_pipes" />) in this example. These objects establish communication between the orchestration and external process. On the orchestration end, these match a corresponding `PipesContextInjector` and `PipesMessageReader`, which are instantiated inside the <PyObject module="dagster_k8s" object="PipesK8sClient" />.
+
+- **Inside the body of the context manager (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), retrieve a log and report an asset materialization.** These calls use the temporary communications channels established by <PyObject object="PipesDefaultContextLoader" module="dagster_pipes" /> and <PyObject object="PipesDefaultMessageWriter" module="dagster_pipes" />. To see the full range of what you can do with the <PyObject object="PipesContext" module="dagster_pipes" />, see the API docs or the general [Pipes documentation](/guides/dagster-pipes).
+
+At this point you can execute the rest of your Kubernetes code as normal, invoking various <PyObject object="PipesContext" module="dagster_pipes" /> APIs as needed.
+
+### Step 1.2: Define and build the container image
+
+Next, you'll package the script into a container image using a `Dockerfile`. For example:
+
+```dockerfile
+FROM python:3.10-slim
+
+pip install dagster-pipes
+
+COPY my_python_script.py .
+```
+
+Then, build the image:
+
+```shell
+docker build -t pipes-example:latest
```

Comment:
my mistake - any instance of `pipes-example:latest` should  be `pipes-example:v1`, using the `latest` tag causes some problems on `kind` so I switched it to `v1` and missed some spots

Reply:
Got it!